File: SwatiSinha/binary_search/minimum_limit_of_balls_in_a_bag.java
class Solution {
    public boolean isValid(int[] nums, int mid, int maxOperations) {
        for (int num : nums) {
            maxOperations -= (num - 1) / mid;
        }
        return maxOperations >= 0;
    }

    public int minimumSize(int[] nums, int maxOperations) {
        int min = 1;
        int max = 0;
        for (int num : nums) {
            max = Math.max(num, max);
        }
        while (min <= max) {
            int mid = (min + max) / 2;
            if (isValid(nums, mid, maxOperations)) {
                max = mid - 1;
            } else {
                min = mid + 1;
            }
        }
        return min;
    }
}

File: AdityaKumar1010707/BinarySearch/minimumLimitOfBallsInABag.cpp
class Solution {
    public:
        int checkVal(int value, vector<int> nums) {
            int count = 0;
            for (int i = 0; i < nums.size(); i++) {
                count += (--nums[i] / value);
            }
            cout<<count<<" for num : " << value <<endl;
            return count;
        }
    
        int minimumSize(vector<int>& nums, int maxOperations) {
            int min = 1;
            int max = 1e9;
          
            while (min < max) {
    
                int mid = min + (max - min) / 2;
                int val = checkVal(mid, nums);
    
                if (val > maxOperations) {
                    // Go right
                    min = mid + 1;
                } else {
                    // Go right
                    
                    max = mid;
                }
            }
    
            return min;
        }
    };